java-lib/src/main/java/com/wavefront/ingester/GraphiteDecoder.java

```diff
@@ -26,19 +27,24 @@
       .appendOptionalTimestamp().whiteSpace()
       .appendAnnotationsConsumer().whiteSpace().build();
   private final String hostName;
+  private LinkedHashSet<String> customSourceTags;
 
-  public GraphiteDecoder() {
+  public GraphiteDecoder(LinkedHashSet<String> customSourceTags) {
```

Comment:
Alright, would you be OK with me switching it to a List? I think the intent of what should happen is then much clearer than Collection with a comment about iteration order mattering.

I only picked LinkedHashSet because I wanted it to attempt to dtrt if a user had repeated stuff in their wavefront.conf.

I think you are right though. It is better to give a warning if someone puts a tag twice than silently fixing it as it probably indicates they should check their config. I think List is therefore better than Set. 

For now I have made it a warning and the program continues but can change that if you like.